libbpfgo/helpers/kernel_features.go

```diff
@@ -0,0 +1,110 @@
+package helpers
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+type KernelConfig map[string]string
+
+// NewKernelConfig simply creates the underlying map used by
+// the KernelConfig, you must use InitKernelConfig to read
+// in the values of the config
+func NewKernelConfig() KernelConfig {
+	return make(map[string]string)
+}
+
+// InitKernelConfig attempts to read the kernel config
+//  into the KernelConfig from either
+// - /proc/config-$(uname -r)
+// - /boot/config.gz
+func (k KernelConfig) InitKernelConfig() error {
+
+	err := k.getBootConfig()
+	if err == nil {
+		return nil
+	}
+
+	err2 := k.getProcGZConfig()
+	if err != nil {
+		return fmt.Errorf("%v %v", err, err2)
+	}
+
```

Comment:
Another option can be when CONFIG_IKHEADERS is defined. In that case, the headers can be found in sysfs

Reply:
This is a good idea, handling this.

Reply:
Actually, I think that I confused you here. When CONFIG_IKHEADERS is set, the headers themselves are available in sysfs, not the config options set for the kernel

Reply:
Ah, I thought you were talking about `CONFIG_IKCONFIG` and therefore suggesting I add functionality to do this:

https://stackoverflow.com/a/45654468

I think I need to explore that more and do it in a follow up PR if needed. It's not done in `bpftool feature`, let me think about why.